spikeextractors/extractors/jrcsortingextractor/jrcsortingextractor.py

```diff
@@ -0,0 +1,96 @@
+        spike_clusters = self._getfield("spikeClusters").ravel()  # uint32
+        spike_amplitudes = self._getfield("spikeAmps").ravel()  # int16
+        spike_sites = self._getfield("spikeSites").ravel()  # uint32
+        spike_positions = self._getfield("spikePositions").T  # float32
+
+        unit_centroids = self._getfield("clusterCentroids").astype(np.float).T
+        unit_sites = self._getfield("clusterSites").astype(np.uint32).ravel()
+        mean_waveforms = self._getfield("meanWfGlobal").T
+        mean_waveforms_raw = self._getfield("meanWfGlobalRaw").T
+
+        # try to extract the sample rate from the .prm file
+        sample_rate = None
+        prm_file = Path(file_path.name.replace("_res.mat", ".prm"))
+        with prm_file.open("r") as fh:
+            line = fh.readline()
+            while line:
+                line = line.split('%')[0].strip(" ;")
+                if line.startswith("sampleRate"):
+                    line = line.split("=", 1)
+                    try:
+                        sample_rate = float(line[1])
+                    except (IndexError, ValueError):
+                        raise AttributeError("Malformed or missing value for sample rate.")
+                    break
+                line = fh.readline()
+
+        if sample_rate is None:
+            raise AttributeError("Sample rate was not defined.")
+
+        self.set_sampling_frequency(sample_rate)
+
+        # traces, features
+        raw_file = Path(file_path.name.replace("_res.mat", "_raw.jrc"))
+        raw_shape = self._getfield("rawShape").ravel().astype(np.int)
+
+        filt_file = Path(file_path.stem + "_filt.jrc")
+        filt_shape = self._getfield("filtShape").ravel().astype(np.int)
+
+        features_file = Path(file_path.stem + "_features.jrc")
+        features_shape = self._getfield("featuresShape").ravel().astype(np.int)
+
+        # nonpositive clusters are noise or deleted units
+        if positive_only:
+            good_mask = spike_clusters > 0
+        else:
+            good_mask = np.ones_like(spike_clusters, dtype=np.bool)
+
+        self._unit_ids = np.unique(spike_clusters[good_mask])
+
+        # load spike trains
+        self._spike_trains = {}
+        for uid in self._unit_ids:
+            mask = (spike_clusters == uid)
+            self._spike_trains[uid] = spike_times[mask]
+
+            self.set_unit_spike_features(uid, "spike_amplitudes", spike_amplitudes[mask])
+            self.set_unit_spike_features(uid, "spike_sites", spike_sites[mask])
+            self.set_unit_spike_features(uid, "spike_positions", spike_positions[mask, :])
+
+            self.set_unit_property(uid, "centroid", unit_centroids[uid - 1, :])
+            self.set_unit_property(uid, "center_site", unit_sites[uid - 1])
+            self.set_unit_property(uid, "mean_waveform", mean_waveforms[:, :, uid - 1])
+            self.set_unit_property(uid, "mean_waveform_raw", mean_waveforms_raw[:, :, uid - 1])

```

Comment:
Also, can you add this line?
`self._kwargs = {'file_path': str(Path(file_path).absolute(), 'keep_good_only': keep_good_only}`

This makes sure that the extractor can be correctly dumped to file and reloaded

Reply:
`file_path` is already in `MatSortingExtractor._kwargs`, but I'll add `keep_good_only`.